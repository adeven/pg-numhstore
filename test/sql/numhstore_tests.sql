CREATE EXTENSION hstore;
CREATE EXTENSION numhstore;

SELECT array_count(ARRAY['de','de','us']);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
SELECT hstore_sub('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
SELECT AVG(store) FROM (VALUES('a=>2, b=4'),('a=>4, b=6 c=>6'::inthstore) )t(store);
SELECT hstore_div('foo=>10, bar => 15'::inthstore, 5);
SELECT hstore_div('foo=>10, bar => 15'::inthstore, 'foo=>5, bar => 3'::inthstore);
SELECT hstore_mul('foo=>10, bar => 15'::inthstore, 5);
SELECT hstore_mul('foo=>10, bar => 15'::floathstore, 2.5);
SELECT hstore_length('foo=>10, bar => 15'::hstore);
SELECT hstore_gt('foo=>10, bar => 15'::hstore,2);
SELECT hstore_sum_up('foo=>10, bar => 15'::inthstore);
SELECT hstore_sum_up('foo=>2.5, bar => 4.5'::floathstore);
SELECT hstore_sum_array(Array['1','1','3'], Array[2,3,5]);