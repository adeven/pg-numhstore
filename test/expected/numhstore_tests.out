CREATE EXTENSION hstore;
CREATE EXTENSION numhstore;
SELECT array_count(ARRAY['de','de','us']);
     array_count      
----------------------
 "de"=>"2", "us"=>"1"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_add('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
             hstore_add             
------------------------------------
 "a"=>"1.3", "b"=>"3.7", "c"=>"2.4"
(1 row)

SELECT hstore_add('a=>1,b=>2'::hstore,'b=>1,c=>2'::inthstore);
          hstore_add          
------------------------------
 "a"=>"1", "b"=>"3", "c"=>"2"
(1 row)

SELECT hstore_sub('a=>2,b=>3'::hstore,'b=>1,c=>2'::inthstore);
          hstore_sub           
-------------------------------
 "a"=>"2", "b"=>"2", "c"=>"-2"
(1 row)

SELECT hstore_sub('a=>1.3,b=>2.5'::hstore,'b=>1.2,c=>2.4'::floathstore);
             hstore_sub              
-------------------------------------
 "a"=>"1.3", "b"=>"1.3", "c"=>"-2.4"
(1 row)

SELECT AVG(store) FROM (VALUES('a=>2, b=4'),('a=>4, b=6 c=>6'::inthstore) )t(store);
ERROR:  Syntax error near '6' at position 8
LINE 1: SELECT AVG(store) FROM (VALUES('a=>2, b=4'),('a=>4, b=6 c=>6...
                                                     ^
SELECT hstore_div('foo=>10, bar => 15'::inthstore, 5);
                        hstore_div                        
----------------------------------------------------------
 "bar"=>"3.0000000000000000", "foo"=>"2.0000000000000000"
(1 row)

SELECT hstore_div('foo=>10, bar => 15'::inthstore, 'foo=>5, bar => 3'::inthstore);
                        hstore_div                        
----------------------------------------------------------
 "bar"=>"5.0000000000000000", "foo"=>"2.0000000000000000"
(1 row)

SELECT hstore_mul('foo=>10, bar => 15'::inthstore, 5);
        hstore_mul        
--------------------------
 "bar"=>"75", "foo"=>"50"
(1 row)

SELECT hstore_mul('foo=>10, bar => 15'::floathstore, 2.5);
          hstore_mul          
------------------------------
 "bar"=>"37.5", "foo"=>"25.0"
(1 row)

SELECT hstore_length('foo=>10, bar => 15'::hstore);
 hstore_length 
---------------
             2
(1 row)

SELECT hstore_gt('foo=>10, bar => 15'::hstore,2);
 hstore_gt 
-----------
 f
(1 row)

SELECT hstore_sum_up('foo=>10, bar => 15'::inthstore);
 hstore_sum_up 
---------------
            25
(1 row)

SELECT hstore_sum_up('foo=>2.5, bar => 4.5'::floathstore);
 hstore_sum_up 
---------------
           7.0
(1 row)

SELECT hstore_sum_array(Array['1','1','3'], Array[2,3,5]);
  hstore_sum_array  
--------------------
 "1"=>"5", "3"=>"5"
(1 row)

